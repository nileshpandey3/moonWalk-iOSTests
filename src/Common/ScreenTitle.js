import React, { Component } from "react";
import styled from "styled-components";
import { Animated, Text, Easing } from "react-native";

const Wrapper = styled.View`
  padding: 0 25px;
  overflow: visible;
  height: 80px;
`;

const MainText = styled.Text`
  font-size: 30px;
  color: #eee;
  font-weight: bold;
  margin-top: 15px;
`;

const BackgroundText = styled(Animated.createAnimatedComponent(Text))`
  position: absolute;
  left: 25px;
  right: 0;
  top: -20px;
  font-size: 100px;
  color: #aaaaaa;
  font-weight: bold;
`;

export default class extends Component {
  state = {
    AppearAnim: new Animated.Value(0)
  };

  componentDidMount() {
    Animated.timing(this.state.AppearAnim, {
      toValue: 1,
      duration: 500,
      easing: Easing.out(Easing.quad)
    }).start();
  }

  render() {
    const { title, noBackgroundText = false, style = {} } = this.props;
    const { AppearAnim } = this.state;
    const left = AppearAnim.interpolate({
      inputRange: [0, 1],
      outputRange: [100, 25]
    });
    const opacity = AppearAnim.interpolate({
      inputRange: [0, 1],
      outputRange: [0, 0.3]
    });

    return (
      <Wrapper style={style}>
        {!noBackgroundText && (
          <BackgroundText
            style={{ left, opacity }}
            numberOfLines={1}
            ellipsizeMode="clip"
          >
            {title}
          </BackgroundText>
        )}
        <MainText>{title}</MainText>
      </Wrapper>
    );
  }
}
