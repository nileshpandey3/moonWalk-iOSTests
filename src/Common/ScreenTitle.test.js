import React from "react";
import ScreenTitle from "./ScreenTitle";
import renderer from "react-test-renderer";

describe("screenTitle", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ScreenTitle title="My screen" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly without background text", () => {
    const tree = renderer
      .create(<ScreenTitle title="My screen" noBackgroundText />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
