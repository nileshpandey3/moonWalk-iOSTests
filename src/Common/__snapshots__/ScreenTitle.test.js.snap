// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screenTitle renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "height": 80,
        "overflow": "visible",
        "paddingBottom": 0,
        "paddingLeft": 25,
        "paddingRight": 25,
        "paddingTop": 0,
      },
      Object {},
    ]
  }
>
  <Text
    ellipsizeMode="clip"
    numberOfLines={1}
    style={
      Object {
        "color": "#aaaaaa",
        "fontSize": 100,
        "fontWeight": "bold",
        "left": 100,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": -20,
      }
    }
  >
    My screen
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#eee",
          "fontSize": 30,
          "fontWeight": "bold",
          "marginTop": 15,
        },
        undefined,
      ]
    }
  >
    My screen
  </Text>
</View>
`;

exports[`screenTitle renders correctly without background text 1`] = `
<View
  style={
    Array [
      Object {
        "height": 80,
        "overflow": "visible",
        "paddingBottom": 0,
        "paddingLeft": 25,
        "paddingRight": 25,
        "paddingTop": 0,
      },
      Object {},
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#eee",
          "fontSize": 30,
          "fontWeight": "bold",
          "marginTop": 15,
        },
        undefined,
      ]
    }
  >
    My screen
  </Text>
</View>
`;
